src/metabase/lib/field.cljc

```diff
@@ -270,6 +270,15 @@
     ;; mostly for the benefit of JS, which does not enforce the Malli schemas.
     "unknown_field"))
 
+(defmethod lib.metadata.calculation/field-id-method :metadata/column
+  [_query _stage-number {field-id :id}]
+  field-id)
+
+(defmethod lib.metadata.calculation/field-id-method :field
+  [query stage-number [_tag _id-or-name, :as field-clause]]
+  (when-let [field-metadata (resolve-field-metadata query stage-number field-clause)]
+    (lib.metadata.calculation/field-id query stage-number field-metadata)))
```

Comment:
Do we have a use case for getting the `field-id` from a Field reference? I'm worried about exposing something like this, because it might temp people writing FE code to use this with an MBQL clause, but it's only safe to use with a `:field` reference clause, so you'd have to know what sort of clause you had to be able to use it, and that goes against the principles of MLv2. I think maybe just a plain 

```clj
(field-id field-metadata) => id
```

would be sufficient for now honestly

Reply:
Not that it affects anything, but the destructuring here is off

```clj
[_tag _id-or-name, :as field-clause]
```

since it's missing the options map.

Reply:
One last thought: it's a little inefficient to resolve the metadata when we have something like `[:field {...} 1]` just so we can turn around and return `1`... metadata resolution is currently free in the FE and in tests, but in the Metric/Segment display name calculation in the backend and for future QP stuff it will hit the application database unless its been previously fetched, so we shouldn't do it when it's unnecessary